spec/services/checksum_validator_spec.rb

```diff
@@ -207,6 +207,24 @@
   end
 
   describe '#validate_checksums' do
+    context 'moab is missing from storage' do
+      before do
+        # fake a moab gone missing by updating the preserved object to use a non-existent druid
+        comp_moab.preserved_object.update(druid: 'tr808sl1200')
```

Comment:
Not a valid druid, spec should fail on invalid format. :-)

Reply:
heh, good catch! (and also 🤦 that my music nerd easter egg wasn't a proper druid afterall -- twas late at night when i wrote this test)

um, will investigate and maybe ticket.  i thought we validated this?  https://github.com/sul-dlss/preservation_catalog/blob/master/app/models/preserved_object.rb#L16

maybe my expectations about AR hooks are off.

Reply:
@julianmorley actually, after looking back at other druids and this one, i tentatively retract my facepalm =)

what's invalid about this one?  2 letters, 3 numbers, 2 letters, 4 numbers.  no vowels.  what's the salient difference from e.g. `bz514sm9647` below?

Reply:
for a moment i was thinking the third section is supposed to be 3 letters, but it is supposed to be 2, right?  the canonical `DruidTools::Druid.pattern` is what's used to validate.

Reply:
no L

Reply:
strict druids don't allow vowels or L

Reply:
https://github.com/sul-dlss/druid-tools/blob/master/lib/druid_tools/druid.rb#L14-L16

Reply:
ah, didn't realize `L` is disallowed!  thanks.  will change the example here, and will do a bit of investigation into why the validation isn't throwing an error as expected.  will timebox and ticket that if it looks like a rabbit hole.

Reply:
you have to tell it to be strict.  had to do it that way for backwards compatibility.

got an incredible amount of flack from a former developer about "where did these druid requirements come from".

Reply:
Well tbf it's not like we could point to an actual spec. :)

Reply:
ok, i believe i fixed this example to conform to the strict version of the druid pattern, and i left it at that since we purposefully validate with the looser version for backwards compatibility.